### PR TITLE
feat(deps): update to `@typescript-eslint/*` v6

### LINKED ISSUE #946: Used version of @typescript-eslint/utils does not support ESLint v9
### Have you read the Troubleshooting section?

No

### Plugin version

v6.4.0

### ESLint version

v9.13.0

### Node.js version

v20.11.1

### package manager and version

yarn 4.0.2

### Operating system

Ubuntu 24.04.1 LTS

### Bug description

I could not find an issue for this "problem", so please close the issue if it is known already.
So far things seem to work correctly except for the warning message.

The current version of @typescript-eslint/utils does not list ESLint v9 as a supported version. The result is a warning when running `yarn install`:
`eslint is listed by your project with version 9.13.0, which doesn't satisfy what eslint-plugin-testing-library (p7b6a7) and other dependencies request (^7.5.0 || ^8.0.0).`

> yarn explain peer-requirements p7b6a7
```
We have a problem with eslint, which is provided with version 9.13.0.
It is needed by the following direct dependencies of workspaces in your project:

  ✓ eslint-plugin-testing-library@npm:6.4.0 [5c80b] (via ^7.5.0 || ^8.0.0 || ^9.0.0)

However, those packages themselves have more dependencies listing eslint as peer dependency:

  ✘ @typescript-eslint/utils@npm:5.62.0 [149d9] (via ^6.0.0 || ^7.0.0 || ^8.0.0)
```

### Steps to reproduce

run yarn install

### Error output/screenshots

_No response_

### ESLint configuration

Not applicable

### Rule(s) affected

None

### Anything else?

_No response_

### Do you want to submit a pull request to fix this bug?

No

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
First part of migration to latest `typescript-eslint` version

CC/ @G-Rath @JoshuaKGoldberg

---

Closes #946

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,7 +51,8 @@
 		"type-check": "tsc --noEmit"
 	},
 	"dependencies": {
-		"@typescript-eslint/utils": "^5.62.0"
+		"@typescript-eslint/scope-manager": "^6.21.0",
```
Comment (G-Rath) [inline package.json:54]: this doesn't seem to be directly used anywhere?

**Review by G-Rath — APPROVED** [review]
looks good to me!

File: package.json (hunk 1 above)
Comment (MichaelDeBoey) [reply package.json:54]: It is in `lib/node-utils/index.ts` as `@typescript-eslint/utils` doesn't export `TSESLintScope` anymore as of v6

**Review by Belco90 — APPROVED** [review]
LGTM, thanks for taking care of this!

**Comment by Belco90** [comment]
All tests are failing, can you take a look @MichaelDeBoey ?

**Review by Belco90 — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -51,7 +51,8 @@
...
+		"@typescript-eslint/utils": "^6.21.0"
```
Comment (jcoyne) [inline package.json:55]: Shouldn't this be:
```
"@typescript-eslint/utils": "^8.12.0"
```

?

Comment (G-Rath) [reply package.json:55]: No, because this pull request is upgrading them to v6, not v8

Comment (jcoyne) [reply package.json:55]: Ah! I see there is a https://github.com/testing-library/eslint-plugin-testing-library/pull/947 v7 and presumably v8 will happen at some point.

**Pull request merged**